examples/symfony.php

```diff
@@ -45,6 +46,13 @@ public function json()
 
         return print_r($payload, true);
     }
+
+    public function valuesArray()
+    {
+        $name = $this->queryParameter('arr')->commaSeparated()->int();
```

Comment:
Generally agreed. 

> I guess here we should also provide a `required` and `defaultsTo`, e.g. `$this->queryParameter('arr')->commaSeparated()->int()->required()`

The problem with the current design is that `->commaSeparated()->int()` _et al._ do not return some instance of AbstractValueParser, but the final value. The design here is a bit tricky, because on the one hand it spares us from creating seven new parser classes (for each type/array), but in return we have a parser class that plays as the endpoint of its requests ([See here](https://github.com/mpscholten/request-parser/blob/3e7913dc3768c3bbb4f9ecae84b3c801b0a9ef99/src/CommaSeparatedTypeParser.php#L21-L28)).

I am leaning towards the multiple classes for each array type, what do you think? That would be the best way to maintain the library as consistent as possible in terms of capabilities.